consult-register.el

```diff
@@ -252,10 +253,15 @@ This function is derived from `register-read-with-preview'."
 	 (help-chars (seq-remove #'get-register (cons help-char help-event-list))))
     (unwind-protect
         (while (not reg)
-	  (while (memq (read-key (propertize (caddr (assq action action-list))
-                                             'face 'minibuffer-prompt))
+	  (while (memq (setq key
+                             (read-key (propertize (caddr (assq action action-list))
+                                                   'face 'minibuffer-prompt)))
 		       help-chars)
             (funcall preview))
+          (when (and (eql key ?\e) (characterp last-input-event))
+            ;; in terminal Emacs M-letter is read as two keys, ESC and the letter,
+            ;; turn that into what would have been read in graphical Emacs
+            (setq last-input-event (logior #x8000000 last-input-event)))
```

Comment:
I am not comfortable with overwriting a variable which I don't own. Please introduce a separate variable here for the key decoding.

Reply:
Will do.

Reply:
It looks better with the new variable anyway.

Reply:
I forgot to reindent, wait a second.

Reply:
OK, should be good now.